### ISSUE TITLE
Notifications Stop working

### ISSUE THREAD

**Issue by creativetrendsapps** [indywidualny/FaceSlim#144]
This could be Samsung related but the notification service stops when swiping the app away from recents.

**Comment by indywidualny** [comment]
It stops for a second to be relaunched immediately. It's normal. It shouldn't stop permanently. If it's not restarted something is really wrong.

**Comment by creativetrendsapps** [comment]
Yes, it's stopped until I open the app again. This only happens on my Samsung device any other device works fine. Damn Samsung!

**Comment by indywidualny** [comment]
:(

**Comment by creativetrendsapps** [comment]
I just got a random notification though. I think it's an issue with Samsung's ram management.

**Comment by indywidualny** [comment]
I'm thinking about using an Alarm Manager to start the service automatically every x minutes. It may be a workaround. Anyway, it's just a thought.

**Comment by creativetrendsapps** [comment]
I was actually about to work out the same thing. I'm using the more polished notification system you implemented, but I think an alarm manger would work as well.

**Comment by creativetrendsapps** [comment]
Fixed on my end:

https://gist.github.com/creativetrendsapps/6332c82276297082f25b

**Comment by creativetrendsapps** [comment]
Also in the MainActivity

if (preferences.getBoolean("notifications_activated", false) || preferences.getBoolean("messages_activated", false)) {
                FolioReceiver.scheduleAlarms(getApplicationContext(), false);
            }

**Comment by indywidualny** [comment]
So it appears that the service is not needed at all. We can as well check notifications inside onReceive() of some receiver periodically invoked by an Alarm Manager. I was working with an Alarm Manager lately and and noticed that a wake lock is needed to be 100% sure that the code inside the receiver will be run. Otherwise the device may go to sleep before everything is done.

**Comment by creativetrendsapps** [comment]
With the alarm manager everything seems to work 100%. Although I didn't include the wake lock feature.

**Comment by indywidualny** [comment]
Since there were no complaints I will leave it this way for now. But it stays open because AlarmManager is the right way to do it.

**Comment by creativetrendsapps** [comment]
Here you go:

https://github.com/creativetrendsapps/FolioforFacebook/blob/master/app/src/main/java/com/creativtrendz/folio/notifications/FolioReceiver.java

**Comment by indywidualny** [comment]
Yay. Your app is opensource now :) Thanks.

pon., 13.06.2016, 14:25 użytkownik Jorell Rutledge <notifications@github.com>
napisał:

**Comment by creativetrendsapps** [comment]
Then in the main activity: if (preferences.getBoolean("notifications_activated", false) || preferences.getBoolean("messages_activated", false)) {
                FolioReceiver.scheduleAlarms(getApplicationContext(), false);
            }
